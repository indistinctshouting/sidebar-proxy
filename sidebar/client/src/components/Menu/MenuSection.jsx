import React from 'react';
import MenuItem from './MenuItem.jsx'

const MenuSection = (props) => {
  return (
    <div>
      <h4>
        {props.section}
      </h4>
      {props.items.map(item => 
        <MenuItem 
          key={item.item_name} 
          details={item}
        />
      )}
      <br></br>
    </div>
  )
}

export default MenuSection;