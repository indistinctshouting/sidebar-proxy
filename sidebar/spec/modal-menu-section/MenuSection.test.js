import React from 'react';
import { shallow } from 'enzyme';
import MenuSection from '../../client/src/components/MenuSection';

describe('MenuSection', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      < MenuSection
        section={'potatoes'}
        items={[{item_name: 1}, {item_name: 2}, {item_name: 3}]}
      />
    )
  })

  it('should render menu component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an h4 tag with the section name', () => {
    let sectionName = wrapper.find('h4');
    expect(sectionName.text()).toMatch('potatoes');
  });

  it('should have as many MenuItems as the length of items', () => {
    let sections = wrapper.find('MenuItem');
    expect(sections.length).toEqual(3);
  });

});