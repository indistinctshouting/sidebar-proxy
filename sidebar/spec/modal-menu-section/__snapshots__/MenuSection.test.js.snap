// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSection should render menu component correctly 1`] = `
<div>
  <h4>
    potatoes
  </h4>
  <MenuItem
    details={
      Object {
        "item_name": 1,
      }
    }
    key="1"
  />
  <MenuItem
    details={
      Object {
        "item_name": 2,
      }
    }
    key="2"
  />
  <MenuItem
    details={
      Object {
        "item_name": 3,
      }
    }
    key="3"
  />
  <br />
</div>
`;
